src/foraging_gui/Dialogs.py

```diff
@@ -87,21 +87,21 @@ def __init__(self, MainWindow, parent=None):
         
         self._connectSignalsSlots()
         self.MainWindow=MainWindow
-        self._Laser_1()
-        self._Laser_2()
-        self._Laser_3()
-        self._Laser_4()
+        self._LaserColor_1()
+        self._LaserColor_2()
+        self._LaserColor_3()
+        self._LaserColor_4()
         self._Laser_calibration()
         self._SessionWideControl()
     def _connectSignalsSlots(self):
-        self.Laser_1.currentIndexChanged.connect(self._Laser_1)
-        self.Laser_2.currentIndexChanged.connect(self._Laser_2)
-        self.Laser_3.currentIndexChanged.connect(self._Laser_3)
-        self.Laser_4.currentIndexChanged.connect(self._Laser_4)
-        self.Laser_1.activated.connect(self._Laser_1)
-        self.Laser_2.activated.connect(self._Laser_2)
-        self.Laser_3.activated.connect(self._Laser_3)
-        self.Laser_4.activated.connect(self._Laser_4)
+        self.LaserColor_1.currentIndexChanged.connect(self._LaserColor_1)
+        self.LaserColor_2.currentIndexChanged.connect(self._LaserColor_2)
+        self.LaserColor_3.currentIndexChanged.connect(self._LaserColor_3)
+        self.LaserColor_4.currentIndexChanged.connect(self._LaserColor_4)
+        self.LaserColor_1.activated.connect(self._LaserColor_1)
+        self.LaserColor_2.activated.connect(self._LaserColor_2)
+        self.LaserColor_3.activated.connect(self._LaserColor_3)
+        self.LaserColor_4.activated.connect(self._LaserColor_4)
```

Comment:
In general, avoid copying and pasting code like this. They are redundant, prone to typos, and hard to maintain and extend.

Instead, you can use

```python
laser_idx = [1, 2, 3, 4]

[getattr(self, f"_LaserColor_{i}"])() for i in laser_idx]
[getattr(self, f"LaserColor_{i}").currentIndexChanged.connect(getattr(self, f"_LaserColor_{i}")) for i in laser_idx]
```

What's better, you don't need to create four different callbacks for self._LaserColor_{i}. You can pass the `i` as an argument.

Reply:
Yes, I can make some changes

Reply:
Updated！